src/pocketmine/command/defaults/DayLockCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\command\defaults;
+
+use pocketmine\command\Command;
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\lang\TranslationContainer;
+use pocketmine\level\Level;
+use pocketmine\Player;
+
+class DayLockCommand extends VanillaCommand{
+
+	public function __construct(string $name){
+		parent::__construct(
+			$name,
+			"%pocketmine.command.daylock.description",
+			"%commands.daylock.usage"
+		);
+		$this->setPermission("pocketmine.command.daylock");
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 1){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		if(count($args) < 1 or $args[0] === "true"){
+			foreach($sender->getServer()->getLevels() as $level){
```

Comment:
Why does the command stop and start time in every level rather than just the one the player is in?

Reply:
That's how it was like in TimeCommand.php, i can change it if needed.

Reply:
The rules around commands operating on levels are currently undefined, and adding level arguments increases the overload complexity. Since we have a garbage commands API at the moment, this kind of change is better avoided.

Better to maintain existing behaviour for consistency's sake if nothing else.